src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java

```diff
@@ -90,6 +90,10 @@ protected String checkKey(final String key) {
     @SuppressWarnings("unchecked")
     public <T> T get(final @NotNull String aKey, final @NotNull Class<T> type) {
```

Comment:
You should also suppress warnings for "java:S2583" to silence sonar.

Reply:
I decided the remove the ```@NotNull``` annotations instead, because they are misleading (and actually wrong).

Reply:
They are good to enforce correct API usage. IMHO you should just call get with a single argument if you don’t care about the type or a default value!

Reply:
the API docs does have the ```@NonNull``` annotation and it makes sense here. Here in this implementation it's misleading.

Reply:
My IDE complains when the annotations are present on the interface but not on the implementation. I can see your POV, but in this case I would lean strongly towards `@SuppressWarnings`. IMHO it's exactly a case like this where adding this annotation documents that the "misimplementation" of the interface is deliberate, and thus encourages interested future maintainers to dig through commit messages and JIRAs in order to understand the rationale.

Reply:
thanks @kwin  and @jsedding  for your suggestions. I added the additional parameter to ```@SuppressWarning``` and re-added the ```@NotNull``` annotation, but left the comment on top.